core/handler/activation.go

```diff
@@ -236,15 +249,54 @@ func (h *handler) HandleActivation(activation *pb_broker.DeduplicatedDeviceActiv
 		return nil, err
 	}
 
-	metadata := activation.ActivationMetadata
-	metadata.GetLorawan().NwkSKey = &dev.NwkSKey
-	metadata.GetLorawan().DevAddr = &dev.DevAddr
+	metadata.NwkSKey = &dev.NwkSKey
+	metadata.DevAddr = &dev.DevAddr
 	res = &pb.DeviceActivationResponse{
 		Payload:            resBytes,
 		DownlinkOption:     activation.ResponseTemplate.DownlinkOption,
-		ActivationMetadata: metadata,
+		ActivationMetadata: activation.ActivationMetadata,
 		Trace:              activation.Trace,
 	}
 
 	return res, nil
 }
+
+func (h *handler) registerDeviceOnJoin(base *device.Device, activation *pb_broker.DeduplicatedDeviceActivationRequest) (*device.Device, error) {
+	clone := base.Clone()
+	clone.DevID = strings.ToLower(fmt.Sprintf("%s-%s", base.DevID, activation.DevEui.String()))
+	clone.DevEUI = *activation.DevEui
+	clone.Description = fmt.Sprintf("On-Join device registered on %s", time.Now().UTC().Format(time.RFC3339))
```

Comment:
"Registered on join on %s"

Also can we have a more human readable date/time format here? It doesn't have to be parsable; we have created and updated time stamps